### ISSUE TITLE
Encourage users to share power (coefficient) curves with the community

### ISSUE THREAD

**Issue by uvchik** [wind-python/windpowerlib#35]
I would like to encourage users to share power (coefficient) curves with the community.

@wind-python/team 

Do you share this idea?
What would be the easiest way to collect them?
What would be the standards (source, license,...) ?

**Comment by Ludee** [comment]
Hi @uvchik,


> Do you share this idea?

I do share this idea, it was one of the reasons to have the data in the OEP. Thanks for the questions.

> What would be the easiest way to collect them?

Until now, they can be collected in this table: https://openenergy-platform.org/dataedit/view/model_draft/openfred_windpower_powercurve
@christian-rli is writing a template to add a power curve using the OEP-API.

> What would be the standards (source, license,...) ?

We tried to have table with comprehensive information, that can be used for other purposes as well. I would considers this the standard structure for now (which can be complemented or changed if necessary). The hardest thing is that the power curves are stored as arrays ([PostgreSQL](https://www.postgresql.org/docs/current/arrays.html)) with x,y values. 

The license issue is still the biggest barrier. Since the turbine data sheets are not licensed, we cannot simply use an open license. We formulated this issue and it will be checked by a specialized lawyer in a couple of weeks. 
Nevertheless, I would prefer a CC0 license for this kind of work.

**Comment by uvchik** [comment]
> @christian-rli is writing a template to add a power curve using the OEP-API.

Thank you, that sounds good. Is this open for everyone? Otherwise I think we should make it possible to pass a special formated csv-file via a pull request and execute the script from time to time to load the entries into the OEP. I think it would be an additional barrier to get an OEP account first.

I guess we will not get hundreds of data set :smirk: We really have to encourage people to it.

@Ludee Thank you for dealing with the license stuff, seems to be a difficult subject.

**Comment by christian-rli** [comment]
Hi all,
I'm working on a script to upload everything via API (as it's supposed to be done eventually anyway), however I'm struggling with a range of datatype issues. Datatype DOUBLE_PRECISION seems not to be supported by the API, yet (I created a new Issue for this). 
My other problem is pandas. I want to end up with a PostgreSQL ARRAY of DOUBLE_PRECISION in the data base. I can import the dtype DOUBLE_PRECISION from sqlalchemy.dialects.postgresql. Now when I have a csv filled with something that looks like a DOUBLE_PRECISION ARRAY I seem to be unable to read_csv from a table and into a dataframe and end up with anything but a string. Also, I can't seem to find an example of how to fill a PostgreSQL ARRAY from a pandas data frame.
Basically the entire script runs from typecast to typecast and there seems to be no native support for any of them. Has anyone worked with PostgreSQL ARRAYS and pandas before? I would very much appreciate any helpful suggestions. Thanks in advance!

**Comment by uvchik** [comment]
Is this script available by now? If so, we should add an description to the windpowerlib documentation.

**Comment by christian-rli** [comment]
Good timing for your question @uvchik ! I just finished a contribution script. We decided to manage the curation of the wind_turbine_library in an issue. https://github.com/OpenEnergyPlatform/data-preprocessing/issues/28 
You are welcome to give feedback on the curation process and make suggestions.

**Comment by birgits** [comment]
Thanks @christian-rli! I also included a section about this in the README and getting_started. @uvchik can we close this issue?

**Comment by uvchik** [comment]
Generally I am fine with the solution but the link in the README-file seems to be broken.

**Comment by birgits** [comment]
The link should work as soon as we have released as the data was moved to the directory oedb.

**Comment by uvchik** [comment]
https://github.com/wind-python/windpowerlib/blob/dev/README.rst#wind-turbine-data

Use the link above and click on the word/link "here" (last line of the paragraph).

**Comment by uvchik** [comment]
Fixed.
